src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/CreateMediaS3BucketEvent.cs

```diff
@@ -14,74 +16,117 @@ namespace OrchardCore.Media.AmazonS3;
-                }
-
-                var bucketRequest = new PutBucketRequest
-                {
-                    BucketName = _options.BucketName,
-                    UseClientRegion = true
-                };
-
-                // Tying to create bucket
-                var response = await _amazonS3Client.PutBucketAsync(bucketRequest);
-
-                if (!response.IsSuccessful())
-                {
-                    _logger.LogError("Unable to create Amazon S3 Bucket. {Response}", response);
-                    return;
-                }
-
-                // Blocking public access for the newly created bucket.
-                var blockConfiguration = new PublicAccessBlockConfiguration
-                {
-                    BlockPublicAcls = true,
-                    BlockPublicPolicy = true,
-                    IgnorePublicAcls = true,
-                    RestrictPublicBuckets = true
-                };
-
-                await _amazonS3Client.PutPublicAccessBlockAsync(new PutPublicAccessBlockRequest
-                {
-                    PublicAccessBlockConfiguration = blockConfiguration,
-                    BucketName = _options.BucketName
-                });
-
-                _logger.LogDebug("Amazon S3 Bucket {BucketName} created.", _options.BucketName);
+                _logger.LogInformation("Amazon S3 Bucket {BucketName} already exists.", _options.BucketName);
+                return;
             }
-            catch (Exception e)
+
+            var bucketRequest = new PutBucketRequest
             {
-                _logger.LogError(e, "Unable to create Amazon S3 Bucket.");
+                BucketName = _options.BucketName,
+                UseClientRegion = true
+            };
+
+            // Trying to create bucket.
+            var response = await _amazonS3Client.PutBucketAsync(bucketRequest);
+
+            if (!response.IsSuccessful())
+            {
+                _logger.LogError("Unable to create Amazon S3 Bucket {BucketName}", _options.BucketName);
+                return;
             }
+
+            // Blocking public access for the newly created bucket.
+            var blockConfiguration = new PublicAccessBlockConfiguration
+            {
+                BlockPublicAcls = true,
+                BlockPublicPolicy = true,
+                IgnorePublicAcls = true,
+                RestrictPublicBuckets = true
+            };
+
+            await _amazonS3Client.PutPublicAccessBlockAsync(new PutPublicAccessBlockRequest
+            {
+                PublicAccessBlockConfiguration = blockConfiguration,
+                BucketName = _options.BucketName
+            });
+
+            _logger.LogDebug("Amazon S3 Bucket {BucketName} created.", _options.BucketName);
+        }
+        catch (AmazonS3Exception ex)
+        {
+            _logger.LogError(ex, "Unable to create Amazon S3 Bucket.");
+        }
+    }
+
+    public override async Task RemovingAsync(ShellRemovingContext context)
```

Comment:
This is weird to have `RemovingAsync` in `CreateMediaS3BucketEvent`, we could rename the class name to something like `MediaS3BucketEvents`

Reply:
Yes, initially to not add too much diff, but worth to do, so done on my side.